routes/user.js

```diff
@@ -76,28 +82,260 @@ router.post('/library', function(req, res, next) {
+                        }, {
+                            source: s._id
+                        }]
+                    },
+                    function(err, songs) {
+                        if (err) return handleError(err)
+                        translationss.push(songs.map((s) => s))
+                        if (i == arr.length - 1) {
+                            res.render('export1', {
+                                playlistID: playlist._id,
+                                playlistName: playlist.name,
+                                songs: playlist.songs,
+                                translationss: translationss
+                            })
+                        }
+                    })
+            })
+        })
+})
+
+router.post('/playlist/:playlist_name/export1', function(req, res, next) {
+    // console.log(req.body)
+    var playlistID = req.body.playlistID
+    var songID = req.body.songID
+    var translationID = req.body.translationID
+        // console.log(songID)
+        // res.send({msg: 'saving'})
+    ExportSong.findOne({
+        owner: playlistID,
+        song: songID
+    }, function(err, es) {
+        if (err) return handleError(err)
+        console.log(es)
+        es.translations.push(translationID)
+        es.save(function(err) {
+            if (err) {
+                res.status(400).send('error ' + err)
+            } else {
+                res.send({
+                    msg: 'saving done'
+                })
+            }
+        })
+    })
+})
+
+router.delete('/playlist/:playlist_name/export1', function(req, res) {
+    // var playlistID = ObjectId(req.body.playlistID)
+    // var songID = ObjectId(req.body.songID)
+    // var translationID = ObjectId(req.body.translationID)
+    var playlistID = req.body.playlistID
+    var songID = req.body.songID
+    var translationID = req.body.translationID
+    console.log(req.body)
+    ExportSong.findOne({
+            owner: playlistID,
+            song: songID
+        }, function(err, es) {
+            if (err) return handleError(err)
+            var index = es.translations.indexOf(translationID)
+            if (index > -1) {
+                es.translations.splice(index, 1)
+            }
+            es.save(function(err) {
+                if (err) {
+                    res.status(400).send('error ' + err)
+                } else {
+                    res.send({
+                        msg: 'deleting done'
+                    })
+                }
+            })
+        })
+        // ExportSong.update({_id: playlistID}, {
+        //     $pull: {
+        //         translations: translationID
+        //     }
+        // }, {
+        //     multi: true
+        // }, function(err) {
+        //     if (err) return handleError(err)
+        //     res.send({
+        //         msg: 'deleting done'
+        //     })
+        // })
+})
+
+router.get('/playlist/:playlist_name/export2', function(req, res) {
```

Comment:
add comments explaining these routes or give the routes names that are more descriptive.